### ISSUE TITLE
Accordion NVDA reader mode adjustment causes long announcements to be cut off in NVDA reader mode

### ISSUE THREAD

**Issue by hannalaakso** [alphagov/govuk-frontend#2373]
## What

The accordion section button has a [CSS adjustment that potentially improves the NVDA reader mode announcements](https://github.com/alphagov/govuk-frontend/pull/2257/commits/6eb5bf788d3fff5f0e11626637df9cad95839fab) by making NVDA reader mode perceive the button content as one element and announcing it as such, instead of multiple ones that are announced separately. It was implemented after some discussion at the accessibility clinic - see [commit message](https://github.com/alphagov/govuk-frontend/pull/2257/commits/6eb5bf788d3fff5f0e11626637df9cad95839fab) for details.

However, it seems that the CSS adjustment causes long announcements to be cut off in NVDA reader mode, for instance 

```
clickable  heading    level 2  button  collapsed    Providing assisted digital support   , Helping people who don't have the skills or access to use a 
```

(This is on the second accordion on https://govuk-frontend-pr-2257.herokuapp.com/full-page-examples/campaign-page).

The CSS adjustment incidentally has the favourable effect of limiting the width of the focus state so we'd have to handle the width of the focus state in a different manner if we removed this fix. I've prototyped that in JS and it seems to work although I need to do a bit more testing. Reverting the fix also seems to resolve this (rather minor) bug in Windows High Contrast Mode: https://github.com/alphagov/govuk-frontend/issues/2354

I think we'd just like to find out from the original contributor on whether they agree that we should revert the CSS adjustment or if there is something we're not aware of. 

## Why

We don't have evidence that the default NVDA reader mode behaviour that the CSS adjustment overrides is an issue as was discussed at the accessibility clinic. If we get evidence that there is an issue with it, we could investigate that. However, for now it might be safer to remove the CSS adjustment to makes sure long NVDA reader mode announcements aren't cut off.

## Who needs to know about this
Developers

## Done when

- [ ] We've considered what impact removing the CSS adjustment might have by speaking to the GOV.UK Accessibility team
- [ ] We've agreed next actions

**Comment by chris-gds** [comment]
(Draft)

Upon investigation of this issue there have been a few discoveries

NVDA has two types of modes, [Browse Mode](https://www.nvaccess.org/files/nvda/documentation/userGuide.html#BrowseMode) and Focus Mode. 

- Browse Mode (using arrow keys) allows _"complex read-only documents such as web pages are browsed in NVDA using browse mode"_ 

- Focus mode is used [when interacting with an element](https://www.accessibility-developer-guide.com/knowledge/screen-readers/desktop/browse-focus-modes/)

The original intent of the [NVDA investigation](https://github.com/alphagov/govuk-design-system/issues/1706#issuecomment-861495145) was to resolve a disjointed announcement of "Show this section" while in Browse Mode.   

This disjointed issue on NVDA was highlighted at the Accessibility clinic yet noted as _not_ a WCAG fail.  It was also noted as being unclear if this would be an issue to NVDA users or if they would expect this.  [After investigation a technique was found that resolved this, and on balance, we decided to proceed](https://github.com/alphagov/govuk-frontend/pull/2257/commits/6eb5bf788d3fff5f0e11626637df9cad95839fab).

"this section" was added as hidden text next to the new icon and the word "Show".
This was done to provide a balance between the desired Design while providing additional context for users with visual impairments. A user would hear "Show this section" giving additional context instead of just "Show" - _Show what?_  

![image](https://user-images.githubusercontent.com/71266765/135486237-085735a8-a5cb-414d-9756-f969ef67fcdf.png)

The issue raised above highlights a particular behaviour when the user is using NVDA Browse Mode and is navigating through the Accordion.  As they moves through the `button`  the text that is announced is:
 
[Insert to go into browse mode, press down key]

```
clickable  heading    level 2  button  collapsed    Providing assisted digital support , Helping people who don't have the skills or access to use a 
```

[press down key again]

```
heading    level 2  button  collapsed    service. , Showthis section
```
The word ```service``` is not read out by NVDA Browse mode in this context and is missed off.

It was unclear why the sentence "Helping people who don't have the skills or access to use a service" was being cut off, on deeper investigation this is related to the user-defined setting while in Browse Mode.  

The number of characters is restricted - in this case to 100.  After reading 100 characters the Content being read will stop and user can press the `down arrow` to read the next section.  In this context the total number of characters inside the entire button is being calculated and as a result, it stops - as expected.  Should a user increase this number to 200 the behaviour is not present.

["Maximum Number of Characters on One Line
This field sets the maximum length of a line in browse mode (in characters)."](https://www.nvaccess.org/files/nvda/documentation/userGuide.html#BrowseMode)

![image](https://user-images.githubusercontent.com/71266765/135487281-8f83eccb-4835-4b28-a287-ae3691410574.png)

In addition, to this should user press the "`B`" key (or `Shift-B`) for "button" NVDA will cycle through the buttons on the page much like headings or the `H` key.  Using the "`B`" will announce the full Content within the button as expected with no cut off to the word `service`

While in browse mode NVDA + Firefox indicates this mode with a red outline while Focus mode is shown with a blue outline (switching from Focus Mode to Browser mode is show with a dotted blue outline)

![Screenshot 2021-09-30 at 16 56 58](https://user-images.githubusercontent.com/71266765/135489887-918e4ffc-f2b8-4685-80d3-37a61f95aff7.png)

![Screenshot 2021-09-30 at 16 56 37](https://user-images.githubusercontent.com/71266765/135489891-6b20c714-d7c5-41c2-9b3b-5340cb9b8862.png)

On balance we've decided to revert this.  

Firstly, it's unclear why the word `service` continues not be announced by NVDA.  Additionally, Focus Mode seems more appropriate for this interaction, and based on NVDA documentation, would expect users to interact with the Accordion in this way.  [Browse Mode seems to be more suitable for reading Content](https://www.accessibility-developer-guide.com/knowledge/screen-readers/desktop/browse-focus-modes/).  Should there be a long piece of content NVDA would appear to intentionally pause for user input rather than read the entire paragraph (based on user settings).  However at present this is unclear if users actually interact with the NVDA in this way on a day-to-day basis and the real-world impact to user from the original disjointed issue.   

Another possible solution is to alter the Design to reflect the hidden text, so instead of just "Show" - "Show this section" is present removing the need for hidden text and the additional CSS technique.  Additionally this solution has the added benefit that what is visually seen matches up what is heard from a screen reader.

Should user testing, feedback or additional information change our understanding of this we can review
